src/Core/Geometry/TopologicalMesh.inl

```diff
@@ -278,6 +280,264 @@ inline void TopologicalMesh::interpolateAllPropsOnFaces(
+
+    LOG( logWARNING ) << "Warning, mesh attribute " << typeid( T ).name()
+                      << " is not supported (only float, vec2, vec3 nor vec4 are supported)";
+    //    static_assert( sizeof( T ) == -1, "this type is not supported" );
+    return {};
+}
+
+template <typename T>
+inline std::vector<std::string>& TopologicalMesh::WedgeCollection::getNameArray() {
+
+    LOG( logWARNING ) << "Warning, mesh attribute " << typeid( T ).name()
+                      << " is not supported (only float, vec2, vec3 nor vec4 are supported)";
+    static_assert( sizeof( T ) == -1, "this type is not supported" );
+    return m_floatAttribNames;
+}
+
+template <typename T>
+inline bool TopologicalMesh::WedgeCollection::setWedgeData( const TopologicalMesh::WedgeIndex& idx,
+                                                            const std::string& name,
+                                                            const T& value ) {
+    if ( idx.isValid() )
+    {
+        auto nameArray = getNameArray<T>();
+        auto itr       = std::find( nameArray.begin(), nameArray.end(), name );
+        if ( itr != nameArray.end() )
+        {
+            auto attrIndex = std::distance( nameArray.begin(), itr );
+            m_data[idx].getWedgeData().getAttribArray<T>()[attrIndex] = value;
+            return true;
+        }
+        else
+        {
+            LOG( logERROR ) << "Warning, set wedge: no wedge attrib named " << name << " of type "
+                            << typeid( T ).name();
+        }
+    }
+    return false;
+}
+
+inline bool
+TopologicalMesh::WedgeCollection::setWedgePosition( const TopologicalMesh::WedgeIndex& idx,
+                                                    const Vector3& value ) {
+    if ( idx.isValid() )
+    {
+        m_data[idx].getWedgeData().m_position = value;
+        return true;
+    }
+    return false;
+}
+
+template <typename T>
+void TopologicalMesh::WedgeCollection::addProp( const std::string& name ) {
+    if ( name != std::string( "in_position" ) ) { getNameArray<T>().push_back( name ); }
+}
+
+inline void TopologicalMesh::WedgeCollection::garbageCollection() {
+    m_data.erase( std::remove_if( m_data.begin(),
+                                  m_data.end(),
+                                  []( const Wedge& w ) { return w.isDeleted(); } ),
+                  m_data.end() );
+}
+
+inline bool TopologicalMesh::WedgeData::operator==( const TopologicalMesh::WedgeData& lhs ) const {
+    return
+        //     m_inputTriangleMeshIndex == lhs.m_inputTriangleMeshIndex &&
+        // m_outputTriangleMeshIndex == lhs.m_outputTriangleMeshIndex &&
+        m_position == lhs.m_position && m_floatAttrib == lhs.m_floatAttrib &&
+        m_vector2Attrib == lhs.m_vector2Attrib && m_vector3Attrib == lhs.m_vector3Attrib &&
+        m_vector4Attrib == lhs.m_vector4Attrib;
+}
+
+// return 1 : equals, 2: strict less, 3: strich greater
+template <typename T>
+int compareVector( const T& a, const T& b ) {
```

Comment:
just be careful, this function will be accessible from outside the scope of this file.